File: 4/04335.py
while 1:
    nums = [True] * 11

    lie = False; exit = False
    while 1:
        n = int(input())
        if not n:
            exit = True
            break

        answer = input()
        if answer == "right on":
            if not nums[n]: lie = True
            break
        elif answer == "too high":
            for i in range(n, 11):
                nums[i] = False
        else:
            for i in range(1, n + 1):
                nums[i] = False

    if exit: break

    if lie: print("Stan is dishonest")
    else: print("Stan may be honest")

File: 4/04335.cpp
#include <iostream>
#include <string>
using namespace std;

int main() {
	bool nums[11];

	while (1) {
		for (int i = 1; i <= 10; i++)
			nums[i] = true;

		bool lie = false;
		bool exit = false;
		while (1) {
			int n; cin >> n;
			getchar();

			if (!n) {
				exit = true;
				break;
			}

			string answer; getline(cin, answer);
			
			if (answer == "right on") {
				lie = !nums[n];
				break;
			}
			else if (answer == "too high") {
				for (int i = n; i <= 10; i++)
					nums[i] = false;
			}
			else {
				for (int i = 1; i <= n; i++)
					nums[i] = false;
			}
		}

		if (exit) break;

		if (lie) cout << "Stan is dishonest\n";
		else cout << "Stan may be honest\n";
	}
}
